### PR TITLE
EIP 721 Using address(0) in place of zero address

### PR DESCRIPTION
**Pull request by godfreyhobbs**
Using `address(0)` makes things much more clear and consistent. `address(0)` is the solidity cast of zero into the address type. The previous use of the term `zero address` makes this document hard to understand.  EIP 721 does not include some magical `zero address`.

When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-X.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your Github username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

**Comment by eip-automerger** [comment]
Hi! I'm a bot, and I wanted to automerge your PR, but couldn't because of the following issue(s):

 - EIP 721 requires approval from one of (nastassia.sachs@protonmail.com, @fulldecent, @dete, @dekz)

**Comment by godfreyhobbs** [comment]
@fulldecent I was not aware of the following:
`@return statements I believe this is user-visible documentation`

Can you provide some more information on how it become `user-visible documentation`?

**Comment by fulldecent** [comment]
https://github.com/ethereum/wiki/wiki/Ethereum-Natural-Specification-Format

screen shot 2018-06-28 at 1 06 17 am

**Comment by godfreyhobbs** [comment]
@fulldecent It looks like `@notice` is user visible.  NOT `@return`.  Anyway, I made the change you suggested.

**Comment by axic** [comment]
An alternate solution is to put in a full length zero address (0x000000000000..000).

**Comment by axic** [comment]
@godfreyhobbs @fulldecent what do you think about the comment above?

**Comment by godfreyhobbs** [comment]
@axic `address(0)` is pretty standard and much more readable.

**Comment by godfreyhobbs** [comment]
@axic @fulldecent Can you take a look at my July 6 comments:
https://github.com/ethereum/EIPs/pull/1171#issuecomment-403231350

**Comment by fulldecent** [comment]
@axic Will `0x0` work in Solidity 0.5.0 as an abbreviation for the long form?

@godfreyhobbs There is one "non-zero" left that requires a little rewording to fix.

> ... require payment when calling `approve` with a non-zero `_approved` if it was previously address(0) ...

@jacqueswww Any complaints here?

@dete @dekz Any comments? Do you support this as is? Would you prefer `0x0` or something else instead? I am inclined to approve this (with the one little fix above) as is.

**Comment by axic** [comment]
> Will 0x0 work in Solidity 0.5.0 as an abbreviation for the long form?

It will not. `address(0)` or `0x00...00` (the long form) will work in both 0.4.x and 0.5.x.

**Comment by godfreyhobbs** [comment]
@axic The following compiles without any address related warnings in remix:

```
contract myContract {
 address foo = address(0);
}
```

Link to remix: https://remix.ethereum.org/#optimize=true&version=soljson-v0.5.0-nightly.2018.10.12+commit.1d312c8e.js

**Comment by axic** [comment]
`address x = 0;` will not work.

**Comment by godfreyhobbs** [comment]
@axic yes the following compiles but it is not very readable.

```
contract myContract {
 address foo = 0x0000000000000000000000000000000000000000;
}              
```

I am inclined to choose `address(0)` over `0x0000000000000000000000000000000000000000`

**Comment by jacqueswww** [comment]
@axic @godfreyhobbs @fulldecent  Sorry to throw a stone in the midst, but I am of the opinion that the wording "zero address" is clearer, and could easier be understood than `address(0)` which is a solidity only term. Also as general rule of thumb, if your documentation starts looking like code - one might consider just leaving the documentation or change the wording of the documentation.

My first prize suggestion would be: use  `zero address (0x0000000000000000000000000000000000000000)` as there is zero ambiguity across all possible platforms (think block explorers / editors / other languages etc.).
Another option is using `zero address (address with all bits set to zero)`.

Second prizes would be something like:
a.) Use `address(0) (0x0000000000000000000000000000000000000000)` on it's first occurrence, then forth use `address(0)`.
b.) Use `zero address (0x0000000000000000000000000000000000000000)` on it's first occurence, the use `zero address` from then on.

I find this PR and the whole problem quite interesting... how would we go about creating a lexicon / glossary lookup table for terms used in EIPs ?

File: EIPS/eip-721.md [hunk 1]
```diff
@@ -130,7 +130,7 @@ interface ERC721 /* is ERC165 */ {
     /// @notice Get the approved address for a single NFT
     /// @dev Throws if `_tokenId` is not a valid NFT.
     /// @param _tokenId The NFT to find the approved address for
-    /// @return The approved address for this NFT, or the zero address if there is none
+    /// @return The approved address for this NFT, or the zero address if there is no approved address
```
Comment (jacqueswww) [inline EIPS/eip-721.md]: If we are changing "zero address", I believe one was missed here?

**Comment by axic** [comment]
I personally don't mind whichever option is decided on and given these are only clarifications in the comments, it should be okay to merge despite it is `Final`. However, at least one of the authors, preferably @fulldecent, would need to approve the change.

**Comment by axic** [comment]
Ping @fulldecent @dete

**Comment by fulldecent** [comment]
The text at controversy are `(from == 0)` and `the zero address`.

Would it be better to say `(the from address is 0)` or `(from is 0)`?

When I think of user documentation I think of my grandmom. If I told my grandmom:

> address(0) indicates...

then she would tell me:

> Will, all this computer stuff has got your brain scrambled. That's not how you use parentheses and don't even get me started on your use of spacing.

**Comment by axic** [comment]
Since this EIP became final since this PR was opened, and one author expressed concerns (https://github.com/ethereum/EIPs/pull/1171#issuecomment-400165347), I suggest to close this PR.

@godfreyhobbs what do you think?

**Pull request closed without merging**